clients/rust/tests/mint.rs

```diff
@@ -117,4 +117,94 @@ mod mint {
+        let mut program_test = create_program_test();
+        program_test.set_compute_max_units(400_000);
+        let mut context = program_test.start_with_context().await;
+
+        // Given a new merkle tree.
+
+        let mut tree_manager = TreeManager::<5, 8>::default();
+        tree_manager.create(&mut context).await.unwrap();
+
+        assert!(find_account(&mut context, &tree_manager.tree.pubkey())
+            .await
+            .is_some());
+
+        // When minting a new cNFT.
+        let owner = Keypair::new();
+
+        let metadata = MetadataArgs {
+            name: String::from("cNFT"),
+            uri: String::from("https://c.nft"),
+            symbol: String::from("cNFT"),
+            creators: vec![Creator {
+                address: context.payer.pubkey(),
+                share: 100,
+                verified: false,
+            }],
+            edition_nonce: None,
+            is_mutable: true,
+            primary_sale_happened: true,
+            seller_fee_basis_points: 500,
+            token_program_version: TokenProgramVersion::Original,
+            token_standard: None,
+            collection: None,
+            uses: None,
+        };
+
+        let minting_result = tree_manager
+            .mint(&mut context, owner.pubkey(), metadata)
+            .await;
+
+        // Minting must fail because the token standard hasn't been provided.
+        assert!(minting_result.is_err());
+    }
+
+    #[tokio::test]
+    async fn mint_wrong_token_standard() {
+        let mut program_test = create_program_test();
+        program_test.set_compute_max_units(400_000);
+        let mut context = program_test.start_with_context().await;
+
+        // Given a new merkle tree.
+
+        let mut tree_manager = TreeManager::<5, 8>::default();
+        tree_manager.create(&mut context).await.unwrap();
+
+        assert!(find_account(&mut context, &tree_manager.tree.pubkey())
+            .await
+            .is_some());
+
+        // When minting a new cNFT.
+        let owner = Keypair::new();
+
+        let metadata = MetadataArgs {
+            name: String::from("cNFT"),
+            uri: String::from("https://c.nft"),
+            symbol: String::from("cNFT"),
+            creators: vec![Creator {
+                address: context.payer.pubkey(),
+                share: 100,
+                verified: false,
+            }],
+            edition_nonce: None,
+            is_mutable: true,
+            primary_sale_happened: true,
+            seller_fee_basis_points: 500,
+            token_program_version: TokenProgramVersion::Original,
+            token_standard: Some(TokenStandard::FungibleAsset),
+            collection: None,
+            uses: None,
+        };
+
+        let minting_result = tree_manager
+            .mint(&mut context, owner.pubkey(), metadata)
+            .await;
+
+        // Minting must fail because the provided token standard is Fungible.
+        assert!(minting_result.is_err());
```

Comment:
Could you also do `assert_eq!(minting_result.unwrap_err(), BubblegumError::InvalidTokenStandard)` to make sure it fails for the exact expected reason?

although you already covered this in the JS test so its ok without it, but might as well.

Reply:
I doubt they could've been compared in a such way because they're not implementing `PartialEq`, could have they? `asset_matches` should've been used instead.